src/snowflake/snowpark/session.py

```diff
@@ -1043,12 +1047,33 @@ def write_pandas(
                id  name
             0   1  Jane
 
+            >>> pandas_df4 = pd.DataFrame([(1, "Jane")], columns=["id", "name"])
+            >>> snowpark_df4 = session.write_pandas(pandas_df4, "write_pandas_transient_table", auto_create_table=True, table_type="transient")
+            >>> snowpark_df4.to_pandas()
+               id  name
+            0   1  Jane
+
         Note:
             Unless ``auto_create_table`` is ``True``, you must first create a table in
             Snowflake that the passed in pandas DataFrame can be written to. If
             your pandas DataFrame cannot be written to the specified table, an
             exception will be raised.
         """
+        if create_temp_table:
+            warnings.warn(
+                "create_temp_table is deprecated. We still respect this parameter when it is True but "
+                'please consider using `table_type="temporary"` instead.',
+                DeprecationWarning,
+                # warnings.warn -> write_pandas
+                stacklevel=2,
+            )
+            table_type = "temporary"
```

Comment:
should we take care of case when someone sets `create_temp_table=True` and `table_type='transient'`. In this case `table_type` will be overwritten to `temporary` (which won't be a problem if it means the same thing as `transient`)

Reply:
I think create_temp_table will just overwrite table_type. Maybe worth mentioning it in docstring and a test?

Reply:
Good point. I'll add this to doc string as well as making the "no breaking change" test to cover this overwrite